aws-opentelemetry-distro/src/amazon/opentelemetry/distro/aws_opentelemetry_configurator.py

```diff
@@ -771,8 +772,24 @@ def _check_emf_exporter_enabled() -> bool:
 
     return True
 
-
 def _create_emf_exporter():
+    """
+    Create the appropriate EMF exporter based on the environment and configuration.
+
+    """
+    log_header_setting = _validate_and_fetch_logs_header()
```

Comment:
`_validate_and_fetch_logs_header()` can be ran again when `_create_cloudwatch_emf_exporter()` executed. Maybe we can update `_create_cloudwatch_emf_exporter` to take in an optional value for `log_header_setting`? e.g. if we provide `log_header_setting` to `_create_cloudwatch_emf_exporter`, then it won't re-fetch the log headers. If not provided, use existing logic.